app/modals/transaction/transaction-modal.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { FC, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+import BN from 'bn.js';
+import { BigNumber } from 'bignumber.js';
+import { Modal, Text, Button } from '@blockstack/ui';
+import { StacksTransaction } from '@blockstack/stacks-transactions';
+
+import { RootState } from '../../store';
+import { validateStacksAddress } from '../../utils/get-stx-transfer-direction';
+
+import { selectTxModalOpen, homeActions } from '../../store/home/home.reducer';
+import { TxModalForm } from './transaction-form';
+import { selectMnemonic } from '../../store/keys';
+import {
+  TxModalHeader,
+  buttonStyle,
+  TxModalFooter,
+  TxModalPreview,
+  TxModalPreviewItem,
+  modalStyle,
+} from './transaction-modal-layout';
+import { createStxTransaction } from '../../crypto/create-stx-tx';
+import { validateAddressChain } from '../../crypto/validate-address-net';
+import { broadcastStxTransaction } from '../../store/transaction';
+import { humanReadableStx, stxToMicroStx } from '../../utils/unit-convert';
+
+interface TxModalProps {
+  balance: string;
+  address: string;
+}
+
+enum TxModalStep {
+  DescribeTx,
+  PreviewAndSend,
+}
+
+type ModalComponents = () => {
+  [component in 'header' | 'body' | 'footer']: JSX.Element;
+};
+
+export const TransactionModal: FC<TxModalProps> = ({ balance, address }) => {
+  const dispatch = useDispatch();
+  const [step, setStep] = useState(TxModalStep.DescribeTx);
+  const [fee, setFee] = useState(new BN(0));
+  const [amount, setAmount] = useState(new BigNumber(0));
+  const [total, setTotal] = useState(new BigNumber(0));
+  const [tx, setTx] = useState<null | StacksTransaction>(null);
+  const [loading, setLoading] = useState(false);
+  const { mnemonic, txModalOpen } = useSelector((state: RootState) => ({
+    txModalOpen: selectTxModalOpen(state),
+    mnemonic: selectMnemonic(state),
+  }));
+
+  const form = useFormik({
+    initialValues: {
+      recipient: '',
+      amount: '',
+    },
+    validationSchema: yup.object().shape({
+      recipient: yup
+        .string()
+        .test('test-is-stx-address', 'Must be a valid Stacks Address', (value = '') =>
+          validateStacksAddress(value)
+        )
+        .test('test-is-for-valid-chain', 'Address is for incorrect network', (value = '') =>
+          validateAddressChain(value)
+        )
+        .test(
+          'test-is-not-my-address',
+          'You cannot send Stacks to yourself',
+          value => value !== address
+        ),
+      amount: yup
+        .number()
+        .positive('You cannot send a negative amount of STX')
+        .typeError('Amount of STX must be described as number')
+        .min(1, 'Smallest transaction is 1 STX')
```

Comment:
Is this intentional?

Reply:
This is an open question: is there a smallest tx? meaning to figure out if I can send 0 STX and just pay the fee.

Reply:
We can get clarification, but I would assume that the smallest amount would be 1 µSTX.

Reply:
Will kill it for now.